### PR TITLE
[tlx] Add new port (Need fixes)

### LINKED ISSUE #10970: [New Port Request] tlx
Library name: tlx

Library description: Collection of Sophisticated C++ Data Structures, Algorithms, and Miscellaneous Helpers

Source repository URL: https://github.com/tlx/tlx

Project homepage (if different from the source repository): http://tlx.github.io/

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by xahon**
- What does your PR fix? (not yet) fixes #10970

Need help. When building project that links to that library an error occurs

1.
> D:\dev\test_tlx\out\build\x64-Release\ninja : error : 'D:/dev/tlx/out/build/x64-Debug (default)/tlx/tlx_debug.lib', needed by 'main.exe', missing and no known rule to make it

I think it is due to `tlx_DIR` variable is being set to `D:/dev/tlx/out/build/x64-Debug (default)` which is kind of invalid path but I don't know where that variable set

2. When creating a shared library (dll) appropriate .lib file is not created

### CODE REVIEW

File: ports/tlx/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        -DTLX_USE_GCOV=off
+        -DTLX_TRY_COMPILE_HEADERS=off
+        -DTLX_MORE_TESTS=off
+        -DTLX_BUILD_STATIC_LIBS=${BUILD_STATIC}
+        -DTLX_BUILD_SHARED_LIBS=${BUILD_SHARED}
+)
+
+vcpkg_install_cmake()
+vcpkg_copy_pdbs()
+vcpkg_fixup_cmake_targets(CONFIG_PATH "CMake/")
```
Comment (NancyLi1013) [inline ports/tlx/portfile.cmake:30]:
```suggestion
if(VCPKG_TARGET_IS_WINDOWS)
    vcpkg_fixup_cmake_targets(CONFIG_PATH "CMake/")
else()
    vcpkg_fixup_cmake_targets(CONFIG_PATH "lib/cmake/tlk")
endif()
```
On unix, the config cmake files are listed in different folder from windows.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/tlx/CONTROL [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Source: tlx
+Version: 05.03.2020
```
Comment (NancyLi1013) [inline ports/tlx/CONTROL:2]: Where do you get the version? I didn't notice this version in https://github.com/tlx/tlx.

File: ports/tlx/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,34 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+# TODO: Fix .dlls not producing .lib files
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO tlx/tlx
+    REF d59c325fb31812047e61aba3d75cc037f92c2b3d
```
Comment (NancyLi1013) [inline ports/tlx/portfile.cmake:7]: The commit id is not the latest released version. We prefer to use a released one `903b9b35df8731496a90d8d74f8bedbad2517d9b`

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/tlx/CONTROL (hunk 2 above)
Comment (xahon) [reply ports/tlx/CONTROL:2]: @NancyLi1013 I used the date of the commit

Comment (NancyLi1013) [reply ports/tlx/CONTROL:2]: Since `tlx ` has released version. We suggest to use a released version.

**Comment by xahon** [comment]
It still throws an error during configuring time

> Could not find a package configuration file provided by "tlx" with any of
> 1> [CMake]   the following names:
> 1> [CMake] 
> 1> [CMake]     tlxConfig.cmake
> 1> [CMake]     tlx-config.cmake

However, `packages\tlx_x86-windows\share\tlx` contains a lot of cmake files including `tlx-config.cmake`

File: ports/tlx/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        -DTLX_BUILD_STATIC_LIBS=${BUILD_STATIC}
+        -DTLX_BUILD_SHARED_LIBS=${BUILD_SHARED}
+)
+
+vcpkg_install_cmake()
+vcpkg_copy_pdbs()
+if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_fixup_cmake_targets(CONFIG_PATH "CMake/")
+else()
+    vcpkg_fixup_cmake_targets(CONFIG_PATH "lib/cmake/tlk")
```
Comment (NancyLi1013) [inline ports/tlx/portfile.cmake:33]:
```suggestion
    vcpkg_fixup_cmake_targets(CONFIG_PATH "lib/cmake/tlx")
```
Sorry, it's my mistake. It should be `tlx`.

**Comment by NancyLi1013** [comment]
@xahon 
You met the error when you used `tlx`?
Have you used it in your project like this?
```
find_package(tlx CONFIG REQUIRED)
target_link_libraries(main PRIVATE tlx)
```

**Review by NancyLi1013 — APPROVED** [review]

File: ports/tlx/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,38 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+# TODO: Fix .dlls not producing .lib files
```
Comment (NancyLi1013) [inline ports/tlx/portfile.cmake:2]: Why can it not support dynamic build?

Comment (xahon) [reply ports/tlx/portfile.cmake:2]: @NancyLi1013 When dll is built there is no symbol resolution `.lib` file generated. Vcpkg prints a warning about it. I decided to make it first static, then someone will make it also able to build shared library in separate pr

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
@xahon 
Is there anything else needed to be done for this PR?

**Pull request merged**